### PR TITLE
Dynamic generator properties added to templater

### PR DESCRIPTION
**Pull request by prakaa**
1. Augments templater to now produce inputs for:
   - Coal prices
   - Gas prices
   - Liquid fuel prices
   - Forced outages
   - Partial outages
   - Seasonal ratings

2. Coal and gas prices required a scenario to be specified. This has been implemented in the config file, with validation of scenarios included
3. Snakecase helper function now properly handles FY year columns

### CODE REVIEW

**Comment by prakaa** [comment]
This is ready for review @nick-gorman

**Comment by nick-gorman** [comment]
Look good @prakaa, just see my queries above.

File: src/ispypsa/templater/nodes.py [hunk 1]
```diff
@@ -51,6 +51,8 @@ def template_nodes(
         }
         template = pd.DataFrame(template, index=[0])
         index_col = "single_region_id"
+    else:
```
Comment (prakaa) [inline src/ispypsa/templater/nodes.py:54]: To delete

File: src/ispypsa/templater/dynamic_generator_properties.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+    ]
+    partial_outages_forecast = partial_outages_forecast.set_index("fuel_type")
+    partial_outages_forecast = _apply_all_coal_averages(partial_outages_forecast)
+    partial_outages_forecast = _convert_financial_year_columns_to_float(
+        partial_outages_forecast.drop(index="All Coal Average")
+    )
+    return partial_outages_forecast
+
+
+def _template_seasonal_ratings(
```
[the remark this answers is not in the document]
Comment (prakaa) [reply src/ispypsa/templater/dynamic_generator_properties.py:172]: Done

File: src/ispypsa/templater/dynamic_generator_properties.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+    )
+    cols = _add_units_to_financial_year_columns(gas_prices.columns)
+    cols[0] = "generator"
+    gas_prices.columns = cols
+    gas_prices = gas_prices.drop(columns="gas_price_scenario").set_index("generator")
+    gas_prices = _convert_financial_year_columns_to_float(gas_prices)
+    return gas_prices
+
+
+def _template_liquid_fuel_prices(parsed_workbook_path: Path | str) -> pd.DataFrame:
```
[the remark this answers is not in the document]
Comment (prakaa) [reply src/ispypsa/templater/dynamic_generator_properties.py]: Done, only keep the config scenario

File: src/ispypsa/templater/dynamic_generator_properties.py [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+    ]
+    seasonal_rating = pd.concat(seasonal_ratings, axis=0)
+    seasonal_rating.columns = [
+        _snakecase_string(col) for col in seasonal_rating.columns
+    ]
+    seasonal_rating = _convert_seasonal_columns_to_float(seasonal_rating)
+    return seasonal_rating
+
+
+def _add_units_to_financial_year_columns(columns: pd.Index) -> list[str]:
```
[the remark this answers is not in the document]
Comment (prakaa) [reply src/ispypsa/templater/dynamic_generator_properties.py:202]: Yes, AEMO has been doing this and generally nice/useful to have units in columns even if it makes it annoying to type/reference in code

File: src/ispypsa/templater/flow_paths.py [hunk 5]
```diff
@@ -24,18 +24,20 @@ def template_flow_paths(
...
     """
     logging.info(f"Creating a flow paths template with {granularity} granularity")
-    validate_granularity(granularity)
     if granularity == "sub_regional":
         template = _template_sub_regional_flow_paths(parsed_workbook_path)
     elif granularity == "regional":
         template = _template_regional_interconnectors(parsed_workbook_path)
     elif granularity == "single_region":
         template = pd.DataFrame()
+    else:
```
[the remark this answers is not in the document]
Comment (prakaa) [reply src/ispypsa/templater/flow_paths.py]: Deleted validation here

File: src/ispypsa/templater/nodes.py (hunk 1 above)
Comment (prakaa) [reply src/ispypsa/templater/nodes.py]: Deleted validation here

**Pull request merged**